pkg/morph/client/client.go

```diff
@@ -69,17 +72,20 @@ type Client struct {
 	// on every normal call.
 	switchLock *sync.RWMutex
 
-	// channel for ws notifications
-	notifications chan rpcclient.Notification
-
-	// channel for internal stop
-	closeChan chan struct{}
+	// channels for ws notifications; protected with switchLock
+	notifications   chan rpcclient.Notification
+	blockRcv        chan *block.Block
+	notificationRcv chan *state.ContainedNotificationEvent
+	notaryReqRcv    chan *result.NotaryRequestEvent
 
 	// cached subscription information
 	subscribedEvents       map[util.Uint160]string
 	subscribedNotaryEvents map[util.Uint160]string
 	subscribedToNewBlocks  bool
```

Comment:
Can we embed `subsInfo` here? It allows us to get rid of `updateSubs` and possibly prevents us from forgetting something to copy when we add it.